src/telemetry/metrics.yaml

```diff
@@ -1776,3 +1697,241 @@ session:
+      silent switches have happened yet.
+    bugs:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1711123
+      - https://mozilla-hub.atlassian.net/browse/VPN-5860
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1711123#c1
+      - https://github.com/mozilla-mobile/mozilla-vpn-client/pull/9057#pullrequestreview-1868746289
+    data_sensitivity:
+      - technical
+    notification_emails:
+      - mcleinman@mozilla.com
+      - vpn-telemetry@mozilla.com
+    expires: 2024-07-15
+
+  changed_to_stable:
+    type: event
+    lifetime: ping
+    send_in_pings:
+      - vpnsession
+      - daemonsession
+    description: |
+      The connection status changed to stable.
+
+      This event is recorded in both main application and daemon.
+      The logic and cadence of health checks for each of these
+      is not the same. On the daemon this checks are only
+      done on the Android daemon and not on the iOS network extension.
+
+      Stable status is when pings are returning in an expected
+      timeframe.
+    bugs:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1711123
+      - https://mozilla-hub.atlassian.net/browse/VPN-5860
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1711123#c1
+      - https://github.com/mozilla-mobile/mozilla-vpn-client/pull/9057#pullrequestreview-1868746289
+    data_sensitivity:
+      - technical
+    notification_emails:
+      - mcleinman@mozilla.com
+      - vpn-telemetry@mozilla.com
+    expires: 2024-07-15
+
+  no_signal_count:
+    type: counter
+    lifetime: ping
+    send_in_pings:
+      - vpnsession
+      - daemonsession
+    description: |
+      Count of times that the connection health check results
+      in no signal.
+
+      The health check counters can be compared in relative terms, but should not be considered as
+      markers of time. There is a race condition in the health check (which calls the telemetry),
+      and we frequently have more than one count per second. This is because we conduct a health check
+      due to the timer timeout, and milliseconds later a ping returns and we do another health check
+      due to that ping returning. 
+    bugs:
+      - https://mozilla-hub.atlassian.net/browse/VPN-5860
+    data_reviews:
+      - https://github.com/mozilla-mobile/mozilla-vpn-client/pull/9057#pullrequestreview-1868746289
+    data_sensitivity:
+      - technical
+    notification_emails:
+      - mcleinman@mozilla.com
+      - vpn-telemetry@mozilla.com
+    expires: 2024-07-15
+
+  unstable_count:
+    type: counter
+    lifetime: ping
+    send_in_pings:
+      - vpnsession
+      - daemonsession
+    description: |
+      Count of times that the connection health check results
+      in unstable.
+
+      The health check counters can be compared in relative terms, but should not be considered as
+      markers of time. There is a race condition in the health check (which calls the telemetry),
```

Comment:
Thought: is this a race condition or just expected behaviour?

Reply:
A bit of both? Because if that network check ping returned before the timer timed out, we'd only have 1 health check. And if the ping never returned, we'd only have 1 health check. This could be considered an expected edge case, so I'm removing the "race condition" language.